src/DoctrineReader.php

```diff
@@ -29,23 +30,43 @@ class DoctrineReader implements CountableReader
      */
     protected $iterableResult;
 
+    /** @var QueryBuilder */
+    protected $queryBuilder;
+
     /**
      * @param ObjectManager $objectManager
-     * @param string        $objectName    e.g. YourBundle:YourEntity
+     * @param string $objectName e.g. YourBundle:YourEntity
      */
     public function __construct(ObjectManager $objectManager, $objectName)
     {
         $this->objectManager = $objectManager;
         $this->objectName = $objectName;
     }
 
+    public function setQueryBuilder(QueryBuilder $queryBuilder)
+    {
+        $this->queryBuilder = $queryBuilder;
+
+        return $this;
+    }
+
+    public function getQueryBuilder()
```

Comment:
This should be private.

Reply:
made as `protected`, as the rest of the properties had this visibility.